src/wp-includes/blocks.php

```diff
@@ -128,44 +175,8 @@ function register_block_script_handle( $metadata, $field_name, $index = 0 ) {
 		return false;
 	}
 
-	// Path needs to be normalized to work in Windows env.
-	static $wpinc_path_norm = '';
-	if ( ! $wpinc_path_norm ) {
-		$wpinc_path_norm = wp_normalize_path( realpath( ABSPATH . WPINC ) );
-	}
-
-	// Cache $template_path_norm and $stylesheet_path_norm to avoid unnecessary additional calls.
-	static $template_path_norm   = '';
-	static $stylesheet_path_norm = '';
-	if ( ! $template_path_norm || ! $stylesheet_path_norm ) {
-		$template_path_norm   = wp_normalize_path( get_template_directory() );
-		$stylesheet_path_norm = wp_normalize_path( get_stylesheet_directory() );
-	}
-
-	$script_path_norm = wp_normalize_path( realpath( dirname( $metadata['file'] ) . '/' . $script_path ) );
-
-	$is_core_block = isset( $metadata['file'] ) && str_starts_with( $metadata['file'], $wpinc_path_norm );
-
-	/*
-	 * Determine if the block script was registered in a theme, by checking if the script path starts with either
-	 * the parent (template) or child (stylesheet) directory path.
-	 */
-	$is_parent_theme_block = str_starts_with( $script_path_norm, trailingslashit( $template_path_norm ) );
-	$is_child_theme_block  = str_starts_with( $script_path_norm, trailingslashit( $stylesheet_path_norm ) );
-	$is_theme_block        = ( $is_parent_theme_block || $is_child_theme_block );
-
-	$script_uri = '';
-	if ( $is_core_block ) {
-		$script_uri = includes_url( str_replace( $wpinc_path_norm, '', $script_path_norm ) );
-	} elseif ( $is_theme_block ) {
-		// Get the script path deterministically based on whether or not it was registered in a parent or child theme.
-		$script_uri = $is_parent_theme_block
-			? get_theme_file_uri( str_replace( $template_path_norm, '', $script_path_norm ) )
-			: get_theme_file_uri( str_replace( $stylesheet_path_norm, '', $script_path_norm ) );
-	} else {
-		// Fallback to plugins_url().
-		$script_uri = plugins_url( $script_path, $metadata['file'] );
-	}
+	$script_path_norm = wp_normalize_path( realpath( $path . '/' . $script_path ) );
+	$script_uri       = get_block_asset_uri( $script_path_norm, $metadata['file'] );
```

Comment:
If name change suggestion above is applied:
```suggestion
	$script_uri       = get_block_asset_url( $script_path_norm, $metadata['file'] );
```